packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-batcher.ts

```diff
@@ -33,6 +36,54 @@ const CallDiagnosticEventsBatcher = Batcher.extend({
       networkType: 'unknown',
     };
 
+    // check event names and append latencies?
+    const eventName = item.eventPayload?.event?.name as MetricEventNames;
+    const joinTimes: ClientEvent['payload']['joinTimes'] = {};
+    const cdl = NewMetrics.callDiagnosticLatencies;
+
+    switch (eventName) {
+      case 'client.interstitial-window.launched':
+        joinTimes.meetingInfoReqResp = cdl.getMeetingInfoReqResp();
+        joinTimes.clickToInterstitial = cdl.getClickToInterstitial();
+        break;
+
+      case 'client.call.initiated':
+        joinTimes.meetingInfoReqResp = cdl.getMeetingInfoReqResp();
+        joinTimes.showInterstitialTime = cdl.getShowInterstitialTime();
+        break;
+
+      case 'client.locus.join.response':
+        joinTimes.meetingInfoReqResp = cdl.getMeetingInfoReqResp();
+        joinTimes.callInitJoinReq = cdl.getCallInitJoinReq();
+        joinTimes.joinReqResp = cdl.getJoinReqResp();
+        joinTimes.joinReqSentReceived = cdl.getJoinRespSentReceived();
+        joinTimes.pageJmt = cdl.getPageJMT();
+        joinTimes.clickToInterstitial = cdl.getClickToInterstitial();
+        joinTimes.interstitialToJoinOK = cdl.getInterstitialToJoinOK();
+        joinTimes.totalJmt = cdl.getTotalJMT();
+        break;
+
+      case 'client.ice.end':
+        joinTimes.ICESetupTime = cdl.getICESetupTime();
+        joinTimes.audioICESetupTime = cdl.getAudioICESetupTime();
+        joinTimes.videoICESetupTime = cdl.getVideoICESetupTime();
+        joinTimes.shareICESetupTime = cdl.getShareICESetupTime();
+        break;
+
+      case 'client.media.rx.start':
+        joinTimes.localSDPGenRemoteSDPRecv = cdl.getLocalSDPGenRemoteSDPRecv();
+        break;
+
+      // TODO: Figure out equivalent for WEBRTC
```

Comment:
is this todo still needed? I thought you've already decided on this one

Reply:
yeah, I still haven't decided on a name and where exactly I should add it. I know roughly where it needs to be. Will be done separately.

Reply:
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-440553